src/components/page-header/change-type-modal.tsx

```diff
@@ -0,0 +1,326 @@
+                    },
+                    {
+                        type: 'select',
+                        label: t('header.settings.procedures.changeLinePathType.changeTo'),
+                        options: availableLineStyleOptions,
+                        value: newLineStyleType,
+                        disabledOptions: ['any', currentLineStyleType],
+                        onChange: value => setNewLinePathType(value as LinePathType),
+                    }
+                );
+        }
+        newFields.push({
+            type: 'switch',
+            label: t('header.settings.procedures.changeColor.title'),
+            isChecked: isColorSwitch,
+            oneLine: true,
+            onChange: value => setIsColorSwitch(value),
+        });
+        if (isColorSwitch) {
+            newFields.push({
+                type: 'switch',
+                label: t('header.settings.procedures.changeColor.any'),
+                isChecked: isCurrentColorAny,
+                oneLine: true,
+                onChange: value => setIsCurrentColorAny(value),
+            });
+            !isCurrentColorAny &&
+                newFields.push({
+                    type: 'custom',
+                    label: t('header.settings.procedures.changeColor.changeFrom'),
+                    component: (
+                        <ThemeButton
+                            theme={currentColor}
+                            onClick={() => {
+                                setIsCurrentThemeRequested(true);
+                                dispatch(openPaletteAppClip(theme));
+                            }}
+                        />
+                    ),
+                });
+            newFields.push({
+                type: 'custom',
+                label: t('header.settings.procedures.changeColor.changeTo'),
+                component: (
+                    <ThemeButton
+                        theme={newColor}
+                        onClick={() => {
+                            setIsNewThemeRequested(true);
+                            dispatch(openPaletteAppClip(theme));
+                        }}
+                    />
+                ),
+            });
+        }
+        setFields(newFields);
+    }, [
+        filter,
+        isStationTypeSwitch,
+        currentStationType,
+        newStationType,
+        isLineStyleTypeSwitch,
+        currentLineStyleType,
+        newLineStyleType,
+        isLinePathTypeSwitch,
+        currentLinePathType,
+        newLinePathType,
+        isColorSwitch,
+        isCurrentColorAny,
+        currentColor,
+        newColor,
+    ]);
+
+    React.useEffect(() => {
+        setIsStationTypeSwitch(false);
+        setIsLineStyleTypeSwitch(false);
+        setIsLinePathTypeSwitch(false);
+        setIsColorSwitch(false);
+    }, [selected]);
+
+    const handleChange = () => {
+        if ((!filter || filter.includes('station')) && isStationTypeSwitch) {
```

Comment:
This is not intuitive as the switch actually unfolds some inputs. `Accordion` from chakra UI could be used. See more in the tools panel.